starport/interface/cli/starport/cmd/network_chain_join.go

```diff
@@ -241,6 +190,115 @@ func networkChainJoinHandler(cmd *cobra.Command, args []string) error {
+	if gentxPath == "" {
+		questions = append(questions,
+			cliquiz.NewQuestion("Staking amount", &proposal.Validator.StakingAmount,
+				cliquiz.DefaultAnswer("95000000stake"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Moniker",
+				&proposal.Validator.Moniker,
+				cliquiz.DefaultAnswer("mynode"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Commission rate",
+				&proposal.Validator.CommissionRate,
+				cliquiz.DefaultAnswer("0.10"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Commission max rate",
+				&proposal.Validator.CommissionMaxRate,
+				cliquiz.DefaultAnswer("0.20"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Commission max change rate",
+				&proposal.Validator.CommissionMaxChangeRate,
+				cliquiz.DefaultAnswer("0.01"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Min self delegation",
+				&proposal.Validator.MinSelfDelegation,
+				cliquiz.DefaultAnswer("1"),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Gas prices",
+				&proposal.Validator.GasPrices,
+				cliquiz.DefaultAnswer("0.025"+denom),
+				cliquiz.Required(),
+			),
+			cliquiz.NewQuestion("Website", &proposal.Meta.Website),
+			cliquiz.NewQuestion("Identity", &proposal.Meta.Identity),
+			cliquiz.NewQuestion("Details", &proposal.Meta.Details),
+		)
+	}
+
+	// prepare questions to interactively ask for a publicAddress when peer isn't provided
+	// and not running through chisel proxy.
+	if publicAddress == "" && !xchisel.IsEnabled() {
+		opts := []cliquiz.Option{
+			cliquiz.Required(),
+		}
+		ip, err := ipify.GetIp()
+		if err == nil {
+			opts = append(opts, cliquiz.DefaultAnswer(fmt.Sprintf("%s:26656", ip)))
+		}
+		questions = append(questions, cliquiz.NewQuestion("Peer's address", &publicAddress, opts...))
+	}
+
+	// interactively ask validator questions if there is a need to collect extra info.
+	if len(questions) > 0 {
+		fmt.Println()
+		printSection("Validator proposal")
+
+		if err := cliquiz.Ask(questions...); err != nil {
+			return nil, "", types.Coin{}, "", err
+		}
+	}
+
+	// issue gentx and return with validator info when gentx isn't provided manually.
+	if gentxPath == "" {
+		if gentx, err = blockchain.IssueGentx(cmd.Context(), *account, proposal); err != nil {
+			return nil, "", types.Coin{}, "", err
+		}
+
+		if selfDelegation, err = types.ParseCoinNormalized(proposal.Validator.StakingAmount); err != nil {
+			return nil, "", types.Coin{}, "", err
+		}
+
+		return gentx, account.Address, selfDelegation, publicAddress, nil
+	}
+
+	// gentx is provided manually so use it and return with validator info.
+	if gentx, err = os.ReadFile(gentxPath); err != nil {
+		return nil, "", types.Coin{}, "", errors.Wrap(err, "cannot open gentx file")
+	}
+
+	info, err := networkbuilder.ParseGentx(gentx)
+	if err != nil {
+		return nil, "", types.Coin{}, "", err
+	}
+	return gentx, info.DelegatorAddress, info.SelfDelegation, publicAddress, nil
```

Comment:
I see that `info.DelegatorAddress` is used as `validatorAddress` for the `AddValidatorProposal`. We must use `info.ValidatorAddress` instead.

Both are the same (because it's a self delegation) but in a different format. The delegator address is an `AccAddress` (`cosmosxxx`) the validator address is a `ValAddress` (`cosmosvalopxxx`).
We must provide the `ValAddress` format to the `AddValidatorProposal` proposal and the `AccAddress` (therefore `info.DelegatorAddress`) tp the `AddAccountProposal`

Reply:
I don't get why we should use `ValAddress` format for `AddValidatorProposal` because we were already using `AccAddress`.

Reply:
> I don't get why we should use ValAddress format for AddValidatorProposal because we were already using AccAddress.

Sorry you're right, I haven't realized we convert it to validator format in `spn` package

https://github.com/tendermint/starport/blob/495e9abfc527d94e2c783186aa78c309ac220664/starport/pkg/spn/proposal.go#L242